rest/src/main/java/discord4j/rest/service/ApplicationService.java

```diff
@@ -32,4 +39,58 @@ public ApplicationService(Router router) {
                 .exchange(getRouter())
                 .bodyToMono(ApplicationInfoData.class);
     }
+
+    public Flux<ApplicationCommandData> getGlobalApplicationCommands(long applicationId) {
+        return Routes.GLOBAL_APPLICATION_COMMANDS_GET.newRequest(applicationId)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData[].class)
+            .flatMapMany(Flux::fromArray);
+    }
+
+    public Mono<ApplicationCommandData> createGlobalApplicationCommand(long applicationId, ApplicationCommandRequest request) {
+        return Routes.GLOBAL_APPLICATION_COMMANDS_CREATE.newRequest(applicationId)
+            .body(request)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData.class);
+    }
+
+    public Mono<ApplicationCommandData> modifyGlobalApplicationCommand(long applicationId, long commandId, ApplicationCommandRequest request) {
+        return Routes.GLOBAL_APPLICATION_COMMAND_MODIFY.newRequest(applicationId, commandId)
+            .body(request)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData.class);
+    }
+
+    public Mono<Void> deleteGlobalApplicationCommand(long applicationId, long commandId) {
+        return Routes.GLOBAL_APPLICATION_COMMAND_DELETE.newRequest(applicationId, commandId)
+            .exchange(getRouter())
+            .bodyToMono(Void.class);
+    }
+
+    public Flux<ApplicationCommandData> getGuildApplicationCommands(long applicationId, long guildId) {
+        return Routes.GUILD_APPLICATION_COMMANDS_GET.newRequest(applicationId, guildId)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData[].class)
+            .flatMapMany(Flux::fromArray);
+    }
+
+    public Mono<ApplicationCommandData> createGuildApplicationCommand(long applicationId, long guildId, ApplicationCommandRequest request) {
+        return Routes.GUILD_APPLICATION_COMMANDS_CREATE.newRequest(applicationId, guildId)
+            .body(request)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData.class);
+    }
+
+    public Mono<ApplicationCommandData> modifyGuildApplicationCommand(long applicationId, long guildId, long commandId, ApplicationCommandRequest request) {
+        return Routes.GUILD_APPLICATION_COMMAND_MODIFY.newRequest(applicationId, guildId, commandId)
+            .body(request)
+            .exchange(getRouter())
+            .bodyToMono(ApplicationCommandData.class);
+    }
+
+    public Mono<Void> deleteGuildApplicationCommand(long applicationId, long guildId, long commandId) {
+        return Routes.GUILD_APPLICATION_COMMAND_DELETE.newRequest(applicationId, guildId, commandId)
+            .exchange(getRouter())
+            .bodyToMono(Void.class);
+    }
```

Comment:
Can you move these methods to InteractionService, so we can keep it local to this feature.

![image](https://user-images.githubusercontent.com/1908830/103136336-3d2bf480-469e-11eb-8d4b-61a999a065a2.png)

In the future we'll plan to resolve the inconsistencies, maybe by moving `/oauth2/applications` route to a dedicated OAuth2 class if we cover #715

Reply:
Actually, I'm not 100% on this. Implementations need to use that `/oauth2/applications` endpoint to get the application ID for endpoints, right?

If that's the case, I can now see the reasoning behind locating these methods here. What do you think?